### PR TITLE
Make LargeFileHelper.php faster by avoiding execs as much as possible

### LINKED ISSUE #8405: `stat` is being launched for every file that is scanned.
I just installed NC 13.0.0 and I noticed that it was taking a lot to scan my photos. I attached an `strace` the the scanner and I found out that's it calling `stat` for each file! I'm not talking about the C or PHP function, I'm talking about `coreutils`' `/usr/bin/stat`!

This has a huge performance impact in my server, and looking at the code, I can't figure out why this is tried before using the PHP function `filesize()`. And why it isn't even using `[stat](https://secure.php.net/manual/en/function.stat.php)`? Not to mention that is not calling `escapeshellarg()` either.

### PR DESCRIPTION
**Pull request by marco44**
Should fix nextcloud/server#8405, as per dissussed in the issue

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: lib/private/LargeFileHelper.php [hunk 1]
```diff
@@ -185,14 +185,22 @@ public function getFileSizeNative($filename) {
 	 * @return int
 	 */
 	public function getFileMtime($fullPath) {
-		if (\OC_Helper::is_function_enabled('exec')) {
-			$os = strtolower(php_uname('s'));
-			if (strpos($os, 'linux') !== false) {
-				return $this->exec('stat -c %Y ' . escapeshellarg($fullPath));
+		try {
+			$result= filemtime($fullPath);
```
Comment (juliushaertl) [inline lib/private/LargeFileHelper.php]: Can you adjust the code style here so we have a space between the variable name and the `=`?

File: lib/private/LargeFileHelper.php [hunk 2]
```diff
@@ -185,14 +185,22 @@ public function getFileSizeNative($filename) {
...
+		} catch (\Exception $e) {
+			$result=-1;
```
Comment (juliushaertl) [inline lib/private/LargeFileHelper.php]: Same here.

File: lib/private/LargeFileHelper.php [hunk 3]
```diff
@@ -175,7 +175,7 @@ public function getFileSizeNative($filename) {
 			// returned int as an unsigned integer and put it into a float.
 			return (float) sprintf('%u', $result);
 		}
-		return $result;
+		return  $result;
```
Comment (juliushaertl) [inline lib/private/LargeFileHelper.php]: 1 space is enough. :wink:

**Comment by MorrisJobke** [comment]
Best to be reviewed in the whitespace cleaned diff: https://github.com/nextcloud/server/pull/9490/files?w=1

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
CI failure seems unrelated

**Comment by MorrisJobke** [comment]
> CI failure seems unrelated

Correct.

I would still like to have feedback from @icewind1991 on this one.

**Review by icewind1991 — APPROVED** [review]
Looks fine

**Comment by MorrisJobke** [comment]
Backport in #9656 

@icewind1991 I guess this backport makes sense, right?

**Pull request merged**